lib/lndclient/LndClient.ts

```diff
@@ -76,18 +82,27 @@ class LndClient extends BaseClient {
   }
 
   public connect = async () => {
-    if (!this.isDisabled()) {
+    if (this.isDisabled()) {
+      this.logger.error(`can't connect to lnd. lnd is disabled`);
```

Comment:
I think a `warn` statement is a little more appropriate here since we I don't consider this an error exactly.

Reply:
See above.
calling connect() for a disabled LND should be considered an error. Hence the message.

Reply:
I've thought about this a bit more and I agree actually, but in that case we should throw a `LND_IS_DISABLED` error to let the caller of `connect` know that the method failed and the system is in an unexpected state - right now that's not happening.

But for the case of already being connected, I don't think we should treat it as an error. The state of the lnd client is already in the desired/expected state, but we're unnecessarily calling the method. I think that's exactly the sort of thing that warn statements usually indicate, unexpected behavior that doesn't compromise core functionality.

Again, and as you pointed out, what this method is really doing is verifying that we *can* connect to LND (and that we're subscribed to invoice updates).

Reply:
Well, I disagree. connect() should be called only when we are disconnected, hence it should be an error.

regarding disable throwing an error, I think this is an edge case and the developer will be able to find it if reported as an error or thrown.

feel free to change it after I'm done with the swap.

Reply:
Ok, I will revisit this method another time.